packages/data-stores/src/auth/controls.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { createRegistryControl } from '@wordpress/data';
+import { stringify } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import { wpcomRequest, WpcomClientCredentials } from '../utils';
+import { FetchAuthOptionsAction, FetchWpLoginAction } from './actions';
+import { STORE_KEY } from './constants';
+
+export function createControls( clientCreds: WpcomClientCredentials ) {
+	return {
+		SELECT_USERNAME_OR_EMAIL: createRegistryControl( registry => () => {
+			return registry.select( STORE_KEY ).getUsernameOrEmail();
+		} ),
+		FETCH_AUTH_OPTIONS: async ( { usernameOrEmail }: FetchAuthOptionsAction ) => {
+			const escaped = encodeURIComponent( usernameOrEmail );
+
+			return await wpcomRequest( {
+				path: `/users/${ escaped }/auth-options`,
+				apiVersion: '1.1',
+			} );
+		},
```

Comment:
This pattern proliferating across stores, I'd really love to see the generic control for `wpcom-proxy-request` move forward:

https://github.com/Automattic/wp-calypso/pull/38959

Reply:
Sounds good, would the idea be (if a store needs multiple controls) to spread the generic control into this stores? e.g.

```typescript
import { controls as genericControls } from '../wpcom-request-controls';

export const controls = {
	...genericControls,
	MY_STORE_CONTROL: () => {}
};
```

Or is there another way to share controls across stores?

Reply:
Yes, I believe we'd merge controls. I believe we'll find that all the controls are generic for handling a small set of side-effects, and the specific functionality will move into action creators. This should also make testing much easier and reduce the need for mocking in many places.

Reply:
Sounds good, I do like less mocking :)